src/ToricVarieties/AlgebraicCycles/constructors.jl

```diff
@@ -0,0 +1,196 @@
+```
+"""
+AlgebraicCycle(cc::CohomologyClass) = AlgebraicCycle(toric_variety(cc), polynomial(cc))
+
+
+#################################
+# 3: Addition, subtraction and scalar multiplication
+#################################
+
+function Base.:+(ac1::AlgebraicCycle, ac2::AlgebraicCycle)
+    if toric_variety(ac1) !== toric_variety(ac2)
+        throw(ArgumentError("The algebraic cyclees must be defined on identically the same toric variety."))
+    end
+    return AlgebraicCycle(toric_variety(ac1), polynomial(ac1) + polynomial(ac2))
+end
+
+
+function Base.:-(ac1::AlgebraicCycle, ac2::AlgebraicCycle)
+    if toric_variety(ac1) !== toric_variety(ac2)
+        throw(ArgumentError("The algebraic cyclees must be defined on identically the same toric variety."))
+    end
+    return AlgebraicCycle(toric_variety(ac1), polynomial(ac1) - polynomial(ac2))
+end
+
+
+Base.:*(c::fmpq, ac::AlgebraicCycle) = AlgebraicCycle(toric_variety(ac), coefficient_ring(toric_variety(ac))(c) * polynomial(ac))
+Base.:*(c::Rational{Int64}, ac::AlgebraicCycle) = AlgebraicCycle(toric_variety(ac), coefficient_ring(toric_variety(ac))(c) * polynomial(ac))
+Base.:*(c::T, ac::AlgebraicCycle) where {T <: IntegerUnion} = AlgebraicCycle(toric_variety(ac), coefficient_ring(toric_variety(ac))(c) * polynomial(ac))
+
+
+#################################
+# 4: Intersection product
+#################################
+
+function Base.:*(ac1::AlgebraicCycle, ac2::AlgebraicCycle)
+    if toric_variety(ac1) !== toric_variety(ac2)
+        throw(ArgumentError("The algebraic cyclees must be defined on identically the same toric variety."))
+    end
+    return AlgebraicCycle(toric_variety(ac1), polynomial(ac1) * polynomial(ac2))
+end
+
+
+Base.:^(ac::AlgebraicCycle, p::T) where {T <: IntegerUnion} = AlgebraicCycle(toric_variety(ac), polynomial(ac)^p)
+
+
+########################
+# 5: Equality
+########################
+
+function Base.:(==)(ac1::AlgebraicCycle, ac2::AlgebraicCycle)
+    return toric_variety(ac1) === toric_variety(ac2) && iszero(polynomial(ac1-ac2))
+end
+
+
+######################
+# 6: Display
+######################s
+
+function Base.show(io::IO, ac::AlgebraicCycle)
+    if is_trivial(ac)
+        join(io, "A trivial algebraic cycle on a normal toric variety")
+    end
+    properties_string = String[]
+    coeffs = coefficients(ac)
+    irreds = irreducible_components(ac)
+    indets = gens(chow_ring(toric_variety(ac)))
+    for i in 1:length(irreds)
+        expos = exponents(irreds[i])
+        m = String[]
+        for j in 1:ncols(expos)
+            for k in 1:expos[1,j]
+                push!(m,string(indets[j]))
+            end
+        end
+        if i == 1
+            push!(properties_string, "An algebraic cycle on a normal toric variety given by " * string(coeffs[i]) * "*V(" * join(m,",") * ")")
+        else
+            if coeffs[i] > 0
+                push!(properties_string, "+" * string(coeffs[i]) * "*V(" * join(m,",") * ")")
```

Comment:
This applies in tons of other places: I usually would prefer string interpolation over concatenation. I think it is usually more readable. Well, except for the `join` call here, which I would move outside:
```suggestion
                tmp = join(m,",")
                push!(properties_string, "+$(coeffs[i])V($tmp)")
```

Reply:
Thank you. You indeed mentioned this before, and I just forgot it. Thank you!